src/commands/scanner/rule/list.ts

```diff
@@ -59,58 +61,71 @@ export default class List extends SfdxCommand {
-    let rules = [
-      {
-        name: 'Rule 1',
-        categories: ['Best Practice', 'Code Styling'],
-        rulesets: ['core/best-practice'],
-        languages: ['JS', 'Apex', 'Java'],
-        author: 'Salesforce'
-      },
-      {
-        name: 'Rule 2',
-        categories: ['Security', 'XSS'],
-        rulesets: ['lib/security'],
-        languages: ['JS'],
-        author: 'Doofenshmirtz Evil Inc'
-      }
-    ];
-    return new Promise((res, rej) => {
-      setTimeout(() => {
-        res(rules);
-      }, 2500);
-    });
+  public async run() : Promise<AnyJson> {
+    // We'll have a bunch of promises to load rules from their source files.
+    let pmdPromise = this.getPmdRules();
+
+    // Once all of those promises are done, we'll filter all of the results.
+    return Promise.all([pmdPromise])
+      .then((res : any[]) => {
+        let filteredPmdRules = res[0].filter((rule) => {return this.ruleSatisfiesFilterConstraints(rule);});
+        this.ux.table(filteredPmdRules, columns);
+        return filteredPmdRules;
+      }, (rej : string) => {
+        throw new SfdxError(rej);
+      })
   }
 
-  public async run(): Promise<AnyJson> {
-    const cats = this.flags.category;
-    const rulesets = this.flags.ruleset;
-    const sev = this.flags.severity;
-    const langs = this.flags.language;
-    const author = this.flags.standard ? AuthorFilter.StandardOnly : this.flags.custom ? AuthorFilter.CustomOnly : AuthorFilter.All;
+  private async getPmdRules() : Promise<AnyJson> {
+    // PmdCatalogWrapper is a layer of abstraction between the plugin and PMD, facilitating code reuse and other goodness.
+    return new PmdCatalogWrapper().getCatalog()
+      .then((catalog : {rules}) => {
+        // The catalog has a bunch of information on it, but we only need the 'rules' property.
+        const rules = catalog.rules;
+        // We'll want to do some light tampering with the JSON in order to make sure it displays cleanly on the list.
+        return rules.map((rule : {rulesets : string[]}) => {
+          // If any of the rulesets have a name longer than 20 characters, we'll truncate it to 15 and append ellipses,
+          // so it doesn't overflow horizontally.
+          let truncatedRulesets = rule.rulesets.map((ruleset : string) => {
+            return ruleset.length >= 20 ? ruleset.slice(0, 15) + '...' : ruleset;
+          });
+          const cleanedRule = JSON.parse(JSON.stringify(rule));
+          cleanedRule.rulesets = truncatedRulesets;
+          return cleanedRule;
+        });
+      });
+  }
 
-    // Since loading the rules might take a while, log something at the start so the user doesn't think we're hanging.
-    this.ux.log(messages.getMessage('outputTemplates.preparing'));
+  private ruleSatisfiesFilterConstraints(rule : {categories; rulesets; languages}) : boolean {
```

Comment:
Hm.  I get how this func is used w/r/t PMD above, but it seems like it may not be the most efficient way to query for rules against the catalog.  Because it means you need a list of (all?) rules, and use this to filter them out.  Probably not a problem for us, and if this was an interactive situation where you wanted to change the filters in a UI, this is probably how you would do it (query a long list and then apply different filters on top of it) so perhaps I'll shut up about it.